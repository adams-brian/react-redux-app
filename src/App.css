.loading {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.75;
  background: #cccccc;
}

html, body, #root, .App {
  height: 100%
}

.content {
  flex: 1;
}

.footer {
  height: 50px;
  line-height: 50px;
  background-color: #f5f5f5;
  text-align: center;
}

.counters-container, .about-container {
  text-align: center;
  padding: 1rem;
}

.add-counter, .counter {
  margin: 1rem;
  width: 8rem;
  height: 6rem;
  border: 2px solid #dddddd;
  position: relative;
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
  vertical-align: middle;
}

.add-counter span {
  opacity: 0.5;
  font-size: 1.4rem;
}

.close {
  position: absolute;
  top: 0;
  right: 0;
}

.users-header, .edit-user h1 {
  text-align: center;
  padding: 1rem;
}

.create-user {
  margin-bottom: 1rem;
  display: block;
}
