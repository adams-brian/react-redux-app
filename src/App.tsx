import * as React from 'react';
import { connect } from 'react-redux';
import { Redirect, Route, RouteComponentProps, Switch, withRouter } from 'react-router-dom';

import './App.css';
import About from './common/components/about';
import Footer from './common/components/footer';
import Nav from './common/components/nav';
import Counters from './counters/components/counters';
import Users from './users/components/users';

import { IState } from './store';

interface IAppProps extends RouteComponentProps<{}> {
  loading: boolean;
}

export const App = (props: IAppProps) => (
  <div className="App d-flex flex-column">
    <Nav/>
    <div className="content">
      {props.loading
      ?
        <div className="loading"/>
      :
        <div className="container">
          <Switch>
            <Redirect exact={true} from="/" to="/counters"/>
            <Route path="/counters" component={Counters}/>
            <Route path="/users" component={Users}/>
            <Route path="/about" component={About}/>
          </Switch>
        </div>
      }
    </div>
    <Footer/>
  </div>
);

export default withRouter(
  connect(
    (state: IState) => ({ loading: state.loading })
  )( App )
);
