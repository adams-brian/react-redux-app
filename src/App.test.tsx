import { shallow } from 'enzyme';
import { createLocation, createMemoryHistory } from 'history'
import * as React from 'react';

import { App, mapStateToProps } from './App';
import About from './common/components/about';
import Counters from './counters/components/counters';
import Users from './users/components/users';

describe("App", () => {

  let element: JSX.Element;

  beforeEach(() => {
    element = (
      <App
        history={createMemoryHistory()}
        location={createLocation('/a/location')}
        match={{
          isExact: true,
          params: {},
          path: '/a/location',
          url: 'urlbase/a/location'
        }}
        loading={false}
      />
    )
  });

  it('renders as expected', () => {
    const component = shallow(element);
    expect(component).toMatchSnapshot();
  });

  it('shows loading', () => {
    const component = shallow(
      <App
        history={createMemoryHistory()}
        location={createLocation('/a/location')}
        match={{
          isExact: true,
          params: {},
          path: '/a/location',
          url: 'urlbase/a/location'
        }}
        loading={true}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('routes /counters to Counters', () => {
    const component = shallow(element);
    expect(component.find('Route[path="/counters"]').first().prop('component')).toBe(Counters);
  });

  it('routes /users to Users', () => {
    const component = shallow(element);
    expect(component.find('Route[path="/users"]').first().prop('component')).toBe(Users);
  });

  it('routes /about to About', () => {
    const component = shallow(element);
    expect(component.find('Route[path="/about"]').first().prop('component')).toBe(About);
  });

  it('should map the state to the props', () => {
    expect(mapStateToProps({ loading: false })).toEqual({ loading: false });
    expect(mapStateToProps({ loading: true })).toEqual({ loading: true });
  });
  
});
