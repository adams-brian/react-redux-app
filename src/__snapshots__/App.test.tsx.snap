// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<div
  className="App d-flex flex-column"
>
  <Nav />
  <div
    className="content"
  >
    <div
      className="container"
    >
      <Switch>
        <Redirect
          exact={true}
          from="/"
          push={false}
          to="/counters"
        />
        <Route
          component={[Function]}
          path="/counters"
        />
        <Route
          component={[Function]}
          path="/users"
        />
        <Route
          component={[Function]}
          path="/about"
        />
      </Switch>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`App shows loading 1`] = `
<div
  className="App d-flex flex-column"
>
  <Nav />
  <div
    className="content"
  >
    <div
      className="loading"
    />
  </div>
  <Footer />
</div>
`;
